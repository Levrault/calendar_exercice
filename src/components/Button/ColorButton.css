.button {
  border: 0;
  border-radius: 100%;
  height: resolve($space-m + $space-s);
  outline: none;
  transition: .5s;
  width: resolve($space-m + $space-s);

  &.active,
  &:hover {
    transform: scale(1.2);
  }
}
