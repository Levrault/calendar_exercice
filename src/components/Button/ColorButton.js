import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './ColorButton.css';

/**
 * Color button
 * @param {bool} active
 * @returns {node}
 */
const ColorButton = ({ active, ...rest }) => (
  <button
    type="button"
    className={classnames(styles.container, { [styles.active]: active })}
    style={{ backgroundColor: rest.value }}
    {...rest}
  />
);

ColorButton.propTypes = {
  active: PropTypes.bool
};

export default ColorButton;
