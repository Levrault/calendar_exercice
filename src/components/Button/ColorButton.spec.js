import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from '../../tools/test/enzyme';
import ColorButton from './ColorButton';

const mock = {
  value: '#FF4081',
  onClick: jest.fn()
};

describe('ColorButton component', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <ColorButton
        {...mock}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when activated', () => {
    const component = renderer.create(
      <ColorButton
        {...mock}
        active
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClick function', () => {
    const component = shallow(
      <ColorButton
        {...mock}
      />
    );

    component.simulate('click');
    expect(mock.onClick).toHaveBeenCalled();
  });
});
