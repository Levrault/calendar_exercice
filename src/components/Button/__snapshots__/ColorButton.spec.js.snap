// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorButton component should render correctly 1`] = `
<button
  className=""
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#FF4081",
    }
  }
  type="button"
  value="#FF4081"
/>
`;

exports[`ColorButton component should render correctly when activated 1`] = `
<button
  className="undefined"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#FF4081",
    }
  }
  type="button"
  value="#FF4081"
/>
`;
